enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,325 @@
+scrapes targets as this again can have unforeseen results that can hinder the platform, a very
+low cadence may overwhelm the platform Prometheus instance a very high interval may render some
+of the default alerts ineffective.
+
+The goal of this proposal is to allow users to pick their desired level of scraping while limiting
+the impact this might have on the platform, via resources under the control of the
+cluster-monitoring-operator and other platform operators.
+
+Furthermore to assess the viability of the scrape profile feature the monitoring
+team performed a detailed analysis of its impact in an OpenShift cluster.
+The analysis consisted in a test where an OpenShift cluster would run three replicas
+of the OpenShift Prometheus instance but each replica would be configured to a 
+different scrape profile (`full`, `minimal`, `uponly`). Then we would trigger a 
+workload using kube-burner and at the end of 2 hours, we evaluated the results.
+We concluded that in terms of resource usage, given the results obtained we
+can confidently state that the feature is quite valuable given the reduction of
+CPU by ~21% and memory by ~33% when comparing the `minimal` to the `full` profile.
+More detail can be consulted in this [document](https://docs.google.com/document/d/1MA-HTJQ_X7y_bwpJS2IPbGmC4qDyIMp25jisr34X2F4/edit?usp=sharing)
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to lower the amount of resources consumed by Prometheus in a supported way, so I can choose between different scrape profiles, e.g `full` or `minimal`.
+- As an OpenShift developer, I want a supported way to collect a subset of the metrics exported by my operator depending on the distribution.
+
+### Goals
+
+- Give users a supported and documented method to pick the amount of metrics being consumed by the platform, in a way that allows cluster-monitoring-operator and other operators to provide their own defaulting.
+- Other components of the cluster (console, HPA, alerts) should not be negatively affected by this change.
+- Openshift developers may optionally adhere to this change, developers that do not adhere should not be impacted.
+
+### Non-Goals
+
+- Dynamically adjusting metrics scraped at runtime based on heuristics.
+- Allowing users to explicitly set a list or a regex of metrics that they would
+want to be included in a given profile.
+
+## Proposal
+
+This enhancement leverages label selectors to allow cluster-monitoring-operator to select the appropriate monitors ([Pod|Service]Monitor) for a given scrape profile. 
+
+Given this, the following steps would be necessary to implement this enhancement:
+
+- cluster-monitoring-operator has to be updated to support the logic of the new field;
+- OCP components that would like to implement scrape profiles would have to provide X monitors. Where X is the number of scrape profiles that we would support.
+
+### Workflow Description
```

Comment:
we need to explain what happens if the cluster picks an unsupported scrape profile value (e.g. CMO should probably fail the reconciliation and report a Degraded=True condition).

Reply:
So, that's an interesting point which I hadn't thought about. CMO in on itself (in the current implementation) doesn't run any validation of the scrape profiles that exist. So currently is not able to report a `Degraded=True` condition.
However, this might be an interesting idea to explore as it might also solve https://github.com/openshift/enhancements/pull/1298#discussion_r1067067971 what if CMO would periodically pull all monitors, and then it would:
1. Go through all monitors and extract the services/pods
2. For each service it would either have to find one or len(scrapeProfiles) monitor(s)
3. Either report `Degraded=Flase` or `Degraded=True` if the above check was not successful

Reply:
:+1: